### ISSUE TITLE
Crash when exploding missing anchors

### ISSUE THREAD

**Issue by combor** [mikefarah/yq#474]
YQ 3.3.2 crashes while exploding anchors. This doesnt happen on 3.3.0.

Test YAML:
```
a:
  <<: &anchor
    c: d
  e: f
```

This crashes with:
```
$:tmp$ yq r test.yml -X
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x11fc62a]

goroutine 1 [running]:
github.com/mikefarah/yq/v3/cmd.applyAlias(0xc000186500, 0x0)
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/src/github.com/mikefarah/yq/cmd/utils.go:166 +0x6a
github.com/mikefarah/yq/v3/cmd.explodeNode(0xc000186500, 0x5, 0x127b4bb)
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/src/github.com/mikefarah/yq/cmd/utils.go:214 +0x56f
github.com/mikefarah/yq/v3/cmd.explodeNode(0xc0001863c0, 0x5, 0x127b039)
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/src/github.com/mikefarah/yq/cmd/utils.go:201 +0x58a
github.com/mikefarah/yq/v3/cmd.explode(0xc0001220f0, 0x1, 0x1, 0xc0001220f0, 0x1)
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/src/github.com/mikefarah/yq/cmd/utils.go:232 +0x133
github.com/mikefarah/yq/v3/cmd.printResults(0xc0001220f0, 0x1, 0x1, 0x12d1860, 0xc000122008, 0x0, 0x0)
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/src/github.com/mikefarah/yq/cmd/utils.go:251 +0xa23
github.com/mikefarah/yq/v3/cmd.readProperty(0xc00011ab00, 0xc000134580, 0x1, 0x2, 0x0, 0x0)
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/src/github.com/mikefarah/yq/cmd/read.go:65 +0x176
github.com/spf13/cobra.(*Command).execute(0xc00011ab00, 0xc000134560, 0x2, 0x2, 0xc00011ab00, 0xc000134560)
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0xc00011a840, 0x0, 0x100400, 0xc00010e058)
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
	/private/tmp/yq-20200615-26351-z57dcd/yq-3.3.2/src/github.com/mikefarah/yq/yq.go:11 +0x27
```

but it should produce:
```
a:
  c: d
  e: f
```

**Comment by mikefarah** [comment]
Confirmed - this would have happened when I rewrote explode anchors to be more performant. Will fix.

**Comment by combor** [comment]
Sounds like a proper test is needed for this case.

**Comment by mikefarah** [comment]
Is the 'c' supposed to be indented under the the '<<'? If so, I haven't seen that before and not sure what it means?

**Comment by combor** [comment]
No, sorry, the indentation is wrong here. It should be like this:
```
a:
  <<: &anchor
  c: d
  e: f
```
However I can see this works with my 3.3.0 version now.
```
yq r x -X
a:
  c: d
  e: f
```
```
yq --version 
yq version 3.3.0
```
I think this can be closed

**Comment by mikefarah** [comment]
Cool - in which case this bug has been fixed in 3.4.0
